### PR TITLE
VERIFY-DEPLOYMENT-PROVISION: Add optional vCpu count check

### LINKED ISSUE #438: custom parameter for extra verification in VM provisionining
Recently new VM size test requirement demands the number of vCPU and memory size verification. The VM provision test 'VERIFY-DEPLOYMENT-PROVISION' can take a custom parameter 'vCPU''. If any parameter is set in CustomParameter section in LISAv2, the test will verify the number of vCPU.

### PR DESCRIPTION
**Pull request by stefanlupsa**
Fixes #438 

No custom param (no change in behavior)
```
[LISAv2 Test Results Summary]
Test Run On           : 07/18/2019 14:05:26
ARM Image Under Test  : Canonical : UbuntuServer : 18.04-LTS : latest
Total Test Cases      : 1 (1 Passed, 0 Failed, 0 Aborted, 0 Skipped)
Total Time (dd:hh:mm) : 0:0:6

   ID TestArea             TestCaseName                                                                TestResult TestDuration(in minutes)
-------------------------------------------------------------------------------------------------------------------------------------------
    1 CORE                 VERIFY-DEPLOYMENT-PROVISION                                                       PASS                 2.43
        FirstBoot : PASS
        FirstBoot : Call Trace Verification : PASS
        Reboot : PASS
        Reboot : Call Trace Verification : PASS
```

-CustomTestParameters "VCPU=1" (no override vm size, 1 actual vcpu) 
```
07/18/2019 14:03:32 : [INFO ] Check expected vcpu: 1 actual: 1
[...]
[LISAv2 Test Results Summary]
Test Run On           : 07/18/2019 13:57:42
ARM Image Under Test  : Canonical : UbuntuServer : 18.04-LTS : latest
Total Test Cases      : 1 (1 Passed, 0 Failed, 0 Aborted, 0 Skipped)
Total Time (dd:hh:mm) : 0:0:6

   ID TestArea             TestCaseName                                                                TestResult TestDuration(in minutes)
-------------------------------------------------------------------------------------------------------------------------------------------
    1 CORE                 VERIFY-DEPLOYMENT-PROVISION                                                       PASS                 2.86
        FirstBoot : PASS
        FirstBoot : Call Trace Verification : PASS
        Reboot : PASS
        Reboot : Call Trace Verification : PASS
        vCpu : Check expected vcpu : PASS
```

-CustomTestParameters "VCPU=2" (no override vm size, 1 actual vcpu)
```
07/18/2019 13:55:32 : [ERROR] Check expected vcpu: 2 actual: 1
[...]
[LISAv2 Test Results Summary]
Test Run On           : 07/18/2019 13:50:17
ARM Image Under Test  : Canonical : UbuntuServer : 18.04-LTS : latest
Total Test Cases      : 1 (0 Passed, 1 Failed, 0 Aborted, 0 Skipped)
Total Time (dd:hh:mm) : 0:0:6

   ID TestArea             TestCaseName                                                                TestResult TestDuration(in minutes)
-------------------------------------------------------------------------------------------------------------------------------------------
    1 CORE                 VERIFY-DEPLOYMENT-PROVISION                                                       FAIL                 3.28
        FirstBoot : PASS
        FirstBoot : Call Trace Verification : PASS
        Reboot : PASS
        Reboot : Call Trace Verification : PASS
        vCpu : Check expected vcpu : FAIL
```

-CustomTestParameters "VCPU=8" -OverrideVMSize Standard_A8_v2 (SLES & RHEL)
```
07/18/2019 14:06:27 : [INFO ] Check expected vcpu: 8 actual: 8
[...]
[LISAv2 Test Results Summary]
Test Run On           : 07/18/2019 14:00:43
ARM Image Under Test  : RedHat : RHEL : 7.6 : latest
Total Test Cases      : 1 (1 Passed, 0 Failed, 0 Aborted, 0 Skipped)
Total Time (dd:hh:mm) : 0:0:6

   ID TestArea             TestCaseName                                                                TestResult TestDuration(in minutes)
-------------------------------------------------------------------------------------------------------------------------------------------
    1 CORE                 VERIFY-DEPLOYMENT-PROVISION                                                       PASS                 2.83
        FirstBoot : PASS
        FirstBoot : Call Trace Verification : PASS
        Reboot : PASS
        Reboot : Call Trace Verification : PASS
        vCpu : Check expected vcpu : PASS
```

```
07/18/2019 14:16:16 : [INFO ] Check expected vcpu: 8 actual: 8
[...]
[LISAv2 Test Results Summary]
Test Run On           : 07/18/2019 14:09:47
ARM Image Under Test  : SUSE : SLES : 15 : latest
Total Test Cases      : 1 (1 Passed, 0 Failed, 0 Aborted, 0 Skipped)
Total Time (dd:hh:mm) : 0:0:7

   ID TestArea             TestCaseName                                                                TestResult TestDuration(in minutes)
-------------------------------------------------------------------------------------------------------------------------------------------
    1 CORE                 VERIFY-DEPLOYMENT-PROVISION                                                       PASS                 2.87
        FirstBoot : PASS
        FirstBoot : Call Trace Verification : PASS
        Reboot : PASS
        Reboot : Call Trace Verification : PASS
        vCpu : Check expected vcpu : PASS
```

### CODE REVIEW

File: Testscripts/Windows/VERIFY-DEPLOYMENT-PROVISION.ps1 [hunk 1]
```diff
@@ -6,6 +6,11 @@ param([object] $AllVmData, [object] $CurrentTestData, [object] $TestProvider)
 function Main {
 	param([object] $allVMData, [object] $CurrentTestData, [object] $TestProvider)
 	try {
+		foreach ($param in $currentTestData.TestParameters.param) {
```
Comment (chvalean) [inline Testscripts/Windows/VERIFY-DEPLOYMENT-PROVISION.ps1]: handle the params parse in the Main function call at the end of script, see other ps1.

File: Testscripts/Windows/VERIFY-DEPLOYMENT-PROVISION.ps1 [hunk 2]
```diff
@@ -28,7 +33,31 @@ function Main {
 						-metaData "Reboot : Call Trace Verification" -checkValues "PASS,FAIL,ABORTED" `
 						-testName $currentTestData.testName
 					Write-LogInfo "Test Result : PASS."
-					$testResult = "PASS"
+					# If vCpu parameter is empty, skip the check
+					if ($vCpu -eq "") {
+						Write-LogInfo "Found empty vCpu param, skipping vCpu count check."
+						$testResult = "PASS"
```
Comment (chvalean) [inline Testscripts/Windows/VERIFY-DEPLOYMENT-PROVISION.ps1]: remove this, this is not a pass condition.

File: XML/TestCases/FunctionalTests.xml [hunk 3]
```diff
@@ -4,6 +4,9 @@
         <testScript>VERIFY-DEPLOYMENT-PROVISION.ps1</testScript>
         <files></files>
         <setupType>OneVM</setupType>
+        <TestParameters>
+            <param>vCpu=VCPU</param>
```
Comment (chvalean) [inline XML/TestCases/FunctionalTests.xml:9]: add a comment here that This represents the custom expected number of cores inside the VM.

**Review by chvalean — CHANGES_REQUESTED** [review]

File: XML/TestCases/FunctionalTests.xml (hunk 3 above)
Comment (stefanlupsa) [reply XML/TestCases/FunctionalTests.xml:9]: done

File: Testscripts/Windows/VERIFY-DEPLOYMENT-PROVISION.ps1 (hunk 1 above)
Comment (stefanlupsa) [reply Testscripts/Windows/VERIFY-DEPLOYMENT-PROVISION.ps1]: done

**Review by chvalean — COMMENTED** [review]
changing review to Ju

**Comment by juhlee-microsoft** [comment]
@stefanlupsa I am excited to seeing this PR. Here is one request. As we are reading the test log and test summary, it is nice to print out the VM size if we are using 'VCPU' parameter.

**Review by juhlee-microsoft — APPROVED** [review]

**Pull request merged**